core/src/test/scala/org/apache/spark/SparkContextSuite.scala

```diff
@@ -165,6 +165,17 @@ class SparkContextSuite extends SparkFunSuite with LocalSparkContext with Eventu
     }
   }
 
+  test("add hdfs jar files not exists") {
+    try {
+      val jarPath = "hdfs:///no/path/to/TestUDTF.jar"
+      sc = new SparkContext(new SparkConf().setAppName("test").setMaster("local"))
+      sc.addJar(jarPath)
+      assert(sc.listJars().filter(_.contains("TestUDTF.jar")).size == 0)
```

Comment:
Nit: How about `.forall(j => !j.contains("TestUDTF.jar"))`? or just check `.filter(...).isEmpty`
I guess this is about the best that can be done for a test without an FS to test against. So the behavior change here is that the bad path isn't added.

Reply:
Have changed the test judge code .
Yeah, if path don't add, the error won't happen.